pykafka/balancedconsumer.py

```diff
@@ -535,7 +566,24 @@ def _rebalance(self):
 
                     # Only re-create internal consumer if something changed.
                     if new_partitions != self._partitions:
-                        self._setup_internal_consumer(list(new_partitions))
+                        cns = self._get_internal_consumer(list(new_partitions))
+                        if self._consumer is not None and \
+                                self._partitions_revoked_callback is not None:
+                            self._partitions_revoked_callback(
+                                self, _get_held_offsets_for_partitions(
+                                    current_zk_parts - new_partitions, self._consumer))
+                        if self._partitions_assigned_callback is not None:
+                            new_offsets = self._partitions_assigned_callback(
+                                self,
+                                _get_held_offsets_for_partitions(new_partitions, cns))
```

Comment:
Just took another look at what the JVM consumer docs say (not that we should necessarily do the same), and they pass all partitions here, but they also pass all the old partitions (not just the revoked ones) to the other callback. Maybe that's actually the most flexible way - if you wanted all the old offsets, there's no other way to get them.